sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -505,13 +531,13 @@ extension StorageBlobClient: TransferDelegate {
     }
 
     /// :nodoc:
-    public func uploader(for transfer: BlobTransfer) -> BlobUploader? {
-        transferDelegate?.uploader(for: transfer)
+    public func client(forRestorationId restorationId: String) -> PipelineClient? {
+        transferDelegate?.client(forRestorationId: restorationId)
```

Comment:
Pretty unhappy with these at the moment. Not sure they are in any way better than `uploader/downloader(for transfer)`

Reply:
I removed `options(forRestorationId:...)` so you only get the client. I store the options and the properties as part of the CoreData object, and I'm happy with that for now.

Reply:
Cool, I was thinking about maybe in the future, having the restoration ID be part of the client constructor instead, and keeping a static registry of (restoration ID -> client). Any time a new client is registered, we could see if there are unfulfilled transfers and potentially resume them if their restoration ID exists in the registry. Even if we don't do that last part, we could avoid needing to have the `client(forRestorationId:)` delegate method.

Reply:
I agree. Ideally, I'd like the simplest case to not require any registration (or perhaps automatic registration) and not require this delegate method. For now I'm just focused on having something that works as a baseline for improvement.

Reply:
Agree, we can iterate on this in the future. For now I think it's adding a relatively minimal complexity to make the feature work.